pkg/rtc/room.go

```diff
@@ -342,13 +342,22 @@ func (r *Room) UpdateSubscriptions(
 	subscribe bool,
 ) error {
 	// find all matching tracks
-	tracks := make(map[string]types.PublishedTrack)
+	trackInfos := make(map[string]struct {
+		publisher types.Participant
+		track     types.PublishedTrack
+	})
 	participants := r.GetParticipants()
 	for _, trackSid := range trackIds {
 		for _, p := range participants {
 			track := p.GetPublishedTrack(trackSid)
 			if track != nil {
-				tracks[trackSid] = track
+				trackInfos[trackSid] = struct {
```

Comment:
maybe better to use a a named struct type

Reply:
Yeah, I am not happy with this. I am hoping we can add `RemoveSubscriber` to participant and clean this up.